SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -143,55 +167,31 @@ IPYKERNEL_VERSION=$(python -c "import ipykernel; print(ipykernel.__version__)")
 
 # Check if requirements.txt exists in the repository
 if [ ! -f "${REQ_PATH}" ]; then
-    _log "ERROR: Requirements file not found (${REQ_PATH})."
-    exit 1
-fi
-
-CURRENT_ENV_NAME=$(find "/home/$USER" -type d -name "*_env" | head -n 1 | cut -d '/' -f4)
-CURRENT_REPO_PATH=$(tail -n 1 "/home/$USER/.bash_profile" | cut -d ' ' -f2)
-
-# Check if an environment already exists in the session
-if [ -n "${CURRENT_ENV_NAME}" ]; then
-    _log "ENVIRONMENT_ALREADY_EXISTS:${CURRENT_ENV_NAME}"
-    _log "REPO_PATH:${CURRENT_REPO_PATH#$HOME}"
-    exit 1
+    _error "Requirements file not found (${REQ_PATH})."
 fi
 
-# Create environment (acc-py or generic)
-if [ -n "$ACCPY_VERSION" ]; then
-    source $ACCPY_PATH/base/${ACCPY_VERSION}/setup.sh
-    acc-py venv ${ENV_PATH} | tee -a "${LOG_FILE}"
-else
-    _log "Creating environment ${ENV_NAME} using Generic Python..."
-    python -m venv ${ENV_PATH} | tee -a "${LOG_FILE}"
-fi
-
-_log "ENV_NAME:${ENV_NAME}"
+_log "Creating environment ${ENV_NAME} using ${BUILDER} (${BUILDER_VERSION})..."
+source "${BUILDER_PATH}" | tee -a ${LOG_FILE}
 
 # Make sure the Jupyter server finds the new environment kernel in /home/$USER/.local
 mkdir -p /home/$USER/.local/share/jupyter/kernels
-ln -f -s ${ENV_PATH}/share/jupyter/kernels/${ENV_NAME} /home/$USER/.local/share/jupyter/kernels/${ENV_NAME} | tee -a "${LOG_FILE}"
-
-# Activate the environment
-_log "Setting up the environment..."
-source ${ENV_PATH}/bin/activate
-
-# Install packages in the environment and the same ipykernel that the Jupyter server uses
-_log "Installing packages from ${REQ_PATH}..."
-pip install ipykernel==${IPYKERNEL_VERSION} | tee -a "${LOG_FILE}"
-pip install -r "${REQ_PATH}" | tee -a "${LOG_FILE}"
+ln -f -s ${ENV_PATH}/share/jupyter/kernels/${ENV_NAME} /home/$USER/.local/share/jupyter/kernels/${ENV_NAME} | tee -a ${LOG_FILE}
 
+# Now the environment is set up, and as git repos are installed in /tmp, move it to the $Home/SWAN_projects folder
```

Comment:
I feel this comment is redundant since you already have a similar one inside the if?